### PR TITLE
Add recipe for OptiLCMS

### PR DESCRIPTION
**Pull request by DamienCode404**
## About OptiLCMS
This pull request adds the recipe for the **OptiLCMS** R package to the Bioconda channel.

**OptiLCMS** is an R package designed to streamline the processing of **Liquid Chromatography-Mass Spectrometry (LC-MS)** raw data. It automatically optimizes parameters for better analysis. See the official documentation [here](https://www.metaboanalyst.ca/resources/vignettes/LCMSMS_Raw_Spectral_Processing.html) and the GitHub repository [here](https://github.com/xia-lab/OptiLCMS).

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by DamienCode404** [comment]
@BiocondaBot please add label

File: recipes/bioconductor-optilcms/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+build:
+  number: 0
+  noarch: generic
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  script: 'R CMD INSTALL --build .'
+  run_exports:
+    - '{{ pin_subpackage("bioconductor-optilcms", max_pin="x.x.x") }}'
```
Comment (martin-g) [inline recipes/bioconductor-optilcms/meta.yaml:20]: Are you sure about `x.x.x` ?
See the Pull Request template for an explanation of the meanings - https://github.com/bioconda/bioconda-recipes/blob/master/.github/PULL_REQUEST_TEMPLATE.md#instructions-for-avoiding-api-abi-and-cli-breakage-issues

File: recipes/bioconductor-optilcms/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  summary: "Pipeline for processing LC-MS raw data with optimized parameters."
+  doc_url: https://github.com/xia-lab/OptiLCMS
+  dev_url: https://github.com/xia-lab/OptiLCMS
+
+extra:
+  recipe-maintainers:
+    - DamienCode404
+    - yguitton
+
+channels:
```
Comment (martin-g) [inline recipes/bioconductor-optilcms/meta.yaml:86]: This is not needed.

File: recipes/bioconductor-optilcms/meta.yaml (hunk 1 above)
Comment (DamienCode404) [reply recipes/bioconductor-optilcms/meta.yaml]: Change to `"x"`

**Comment by shang-hongyun** [comment]
e.. How long time will there be a reviewer to check this？

**Comment by DamienCode404** [comment]
@BiocondaBot please review & merge

**Comment by DamienCode404** [comment]
Hello @martin-g, 
This RP has been open for some time now, with the CI/CD stages validated (all checks have passed). Is it possible to merge? Is there anything I need to change? Please let me know what I need to do.
Thank you

**Comment by rpetit3** [comment]
Hi @DamienCode404, please verify the artifacts are working then we can get this merged

**Comment by DamienCode404** [comment]
Hi @rpetit3, 
I have some issues with the installation of a solver backend. Did you know this problem ? I have try some command lines in this [stackoverflow discussion](https://stackoverflow.com/questions/77617946/solve-conda-libmamba-solver-libarchive-so-19-error-after-updating-conda-to-23), but nothing seams to work for me.
Ty !

```
$ conda install -c ./packages optilcms
Error while loading conda entry point: conda-libmamba-solver (libarchive.so.19: cannot open shared object file: No such file or directory)

CondaValueError: You have chosen a non-default solver backend (libmamba) but it was not recognized. Choose one of: classic
```

**Comment by DamienCode404** [comment]
I tried to install with Docker, but my package "optilcms" doesn't seem to be recognized :

```
~/linux-packages$ gzip -dc images/optilcms---1.1.0--r43hdfd78af_0.tar.gz | docker load
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
Getting image source signatures
Copying blob f1df4ffa37e3 skipped: already exists
Copying blob 5f70bf18a086 skipped: already exists
Copying blob 75bcd7b1af8f skipped: already exists
Copying config 36c9f09594 done   |
Writing manifest to image destination
Loaded image: quay.io/biocontainers/optilcms:1.1.0--r43hdfd78af_0
~/linux-packages$ docker images
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
REPOSITORY                      TAG                   IMAGE ID      CREATED     SIZE
quay.io/biocontainers/optilcms  1.1.0--r43hdfd78af_0  36c9f0959443  2 days ago  2.23 GB
(base) laberca@laberca-VirtualBox:~/linux-packages$ docker run -it quay.io/biocontainers/optilcms:1.1.0--r43hdfd78af_0 /bin/bash
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
root@05133aa7e7c6:/# optilcms --help
bash: optilcms: command not found
```

**Comment by DamienCode404** [comment]
I'm not 100% sure, but it looks good to me!
```
~/linux-packages$ docker run -it quay.io/biocontainers/optilcms:1.1.0--r43hdfd78af_0 /bin/bash
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
root@d0f45293ab0f:/# R

R version 4.3.3 (2024-02-29) -- "Angel Food Cake"
Copyright (C) 2024 The R Foundation for Statistical Computing
Platform: x86_64-conda-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(OptiLCMS)
Loading required package: Biobase
Loading required package: BiocGenerics

Attaching package: ‘BiocGenerics’

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min

Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.

Loading required package: BiocParallel
Loading required package: MSnbase
Loading required package: mzR
Loading required package: Rcpp
Loading required package: S4Vectors
Loading required package: stats4

Attaching package: ‘S4Vectors’

The following object is masked from ‘package:utils’:

    findMatches

The following objects are masked from ‘package:base’:

    expand.grid, I, unname

Loading required package: ProtGenerics

Attaching package: ‘ProtGenerics’

The following object is masked from ‘package:stats’:

    smooth


This is MSnbase version 2.28.1
  Visit https://lgatto.github.io/MSnbase/ to get started.


Attaching package: ‘MSnbase’

The following object is masked from ‘package:base’:

    trimws

OptiLCMS 1.1.0 initialized Successfully !
```

**Review by martin-g — APPROVED** [review]

**Pull request merged**